### PR TITLE
Add additional behaviors to Marquee widget.

### PR DESCRIPTION
**Pull request by fxb**
### Description

* Add 'behavior' option:
  - If left empty or set to 'scroll', it results in the existing marquee behavior.
  - If set to 'alternate', the child widget will be scrolled with alternating directions.
  - Attention: 'scroll' does NOT support 'pause_start' and 'pause_midway' options (see below).
  - Attention: 'alternate' does NOT support 'offset_start' and 'offset_end' options.

* Add 'scroll_always' option:
  - If left empty or set to 'False', it results in the existing marquee behavior.
  - If set to 'True', the child widget will be scrolled even if it fits entirely.

* Add 'pause_start' and 'pause_midway' options:
  - Both values default to '1', and have a minumum value of '1'.
  - They control how many frames to pause the animation at the start and at the midway
    point, before changing directions.

* Add tests for all new options.

### Examples

```starlark
behavior = "alternate",
scroll_direction = "horizontal",
```
marquee_horizontal_alternate

```starlark
behavior = "alternate",
scroll_direction = "horizontal",
pause_start = 10,
pause_midway = 10,
```
marquee_horizontal_alternate_pause_10

```starlark
behavior = "alternate",
scroll_direction = "horizontal",
scroll_always = True,
```
marquee_horizontal_alternate_always

```starlark
behavior = "alternate",
scroll_direction = "vertical",
```
marquee_vertical_alternate

```starlark
behavior = "alternate",
scroll_direction = "vertical",
pause_start = 10,
pause_midway = 10,
```
marquee_vertical_alternate_pause_10

```starlark
behavior = "alternate",
scroll_direction = "vertical",
scroll_always = True,
```
marquee_vertical_alternate_always

```starlark
behavior = "scroll",
scroll_direction = "horizontal",
scroll_always = True,
```
marquee_horizontal_scroll_always

### CODE REVIEW

File: render/marquee.go [hunk 1]
```diff
@@ -102,35 +124,140 @@ func (m Marquee) Paint(bounds image.Rectangle, frameIdx int) image.Image {
...
+	if offstart < -cw {
+		offstart = -cw
+	}
+
+	offend := m.OffsetEnd
+	if offend < -cw {
+		offend = -cw
+	}
+
+	return cw + offstart + size - offend + 1
```
Comment (fxb) [inline render/marquee.go:157]: I did not change this, but I actually believe this additional `1` frame is a mistake. If you look at the last example image I posted in this PR, after the 1 frame is repeated at offset 0 and it seems like it "lags".

Comment (matslina) [reply render/marquee.go:157]: I think you may be correct. Can't figure out why it's there.

**Comment by matslina** [comment]
Looks great, thanks fxb!
It does bother me a bit that Marquee is holding so much functionality, but it's also pretty convenient for app devs I'm sure.
:shipit:

**Comment by fxb** [comment]
Yes, I did wonder while adding so much functionality if it would be better to separate it out into two widgets. For example `Marquee` and `AlternateScroll` or so?

**Comment by matslina** [comment]
My idea for a while was to move most of these animations into `render/animation`. See e.g. the unexported (IIRC) `AnimatedPositioned`. Perhaps stuff like `Marquee` and `AlternateScroll` (`Bounce`?) could be implemented as convenient wrappers on top of simpler, animated primitives like that?

**Comment by fxb** [comment]
Yes, I wasn't even aware of `AnimatedPositioned`. I can see if I can implement a `Bounce` off of that one.

**Comment by fxb** [comment]
Ok, new plan is to update this PR to move the new behaviour into a `Bounce` widget as to not increase complexity of `Marquee`. :)

**Comment by fxb** [comment]
Closing in favour of https://github.com/tidbyt/pixlet/pull/213

**Pull request closed without merging**